src/IO/Observer/ReductionActions.hpp

```diff
@@ -163,6 +163,19 @@ namespace ThreadedActions {
  */
 struct WriteReductionData {
  private:
+  static void append_to_reduction_data(
+      gsl::not_null<std::vector<double>*> all_reduction_data, const double t) {
+    all_reduction_data->push_back(t);
+  }
+
+  static void append_to_reduction_data(
+      gsl::not_null<std::vector<double>*> all_reduction_data,
+      const std::vector<double>& t) {
```

Comment:
Thanks for working on observing integrals so quickly! To make sure I understand: Are you planning to use vectors to represent the integral over each component of a tensor? If so, does sending vectors have an advantage over using a `ReductionDatum` with a `double` for each component?

Reply:
Each integral will be `push_back`'d to a `std::vector` containing all the integrals. I actually couldn't make the individual doubles work. Look at #2048 and let me know what you think.